core/amber/src/main/scala/edu/uci/ics/texera/web/resource/aiassistant/AiAssistantResource.scala

```diff
@@ -10,4 +13,42 @@ class AiAssistantResource {
   @RolesAllowed(Array("REGULAR", "ADMIN"))
   @Path("/isenabled")
   def isAiAssistantEnable: String = isEnabled
+
+  /**
+    * To get the type annotation suggestion from OpenAI
+    */
+  @POST
+  @RolesAllowed(Array("REGULAR", "ADMIN"))
+  @Path("/getresult")
```

Comment:
this endpoint `getresult` is not formatted and it is too vague to be meaningful.
maybe consider use `/get-python-type-annotation`  or `/get-type-annotation/python/`

Reply:
after a second read, I understand that this endpoint is designed to be general: you construct prompts in frontend and call this endpoint to get response from LLM. This puts all heavy lifting to frontend (to construct the prompt and decode the response), which is not a good design.

It is better to make this a helper function into the AIAssistantManager. And provide multiple backend apis such as 
- `GET /type-annotation/python?fieldname=xxx` 
And those endpoints will use the helper function in the backend. This way all the prompt generation and decoding will be happening in the backend.